SAGESharp/SLB/IO/BinaryReader.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
```

Comment:
This whole file seems a lot like it is just duplicating the `System.IO.BinaryReader` class.

Reply:
As discussed via chat, we are going to wrap `System.IO.BinaryReader` in `IBinaryReader` rather than a full replacement for testing proposes (mocking framework cannot mock `System.IO.BinaryReader`)